src/sharing/ShareToStream.js

```diff
@@ -77,33 +76,27 @@ class ShareToStreamInner extends React.Component<Props, State> {
   };
 
   handleStreamChange = stream => {
-    this.setState({ stream: stream.trim() });
+    this.setState({ stream });
   };
 
   handleTopicChange = topic => {
-    this.setState({ topic: topic.trim() });
+    this.setState({ topic });
   };
 
   handleStreamAutoComplete = (rawStream: string) => {
     // TODO: What is this for? (write down our assumptions)
     const stream = rawStream.split('**')[1];
-    this.setState({ stream: stream.trim(), isStreamFocused: false });
+    this.setState({ stream, isStreamFocused: false });
   };
 
   handleTopicAutoComplete = (topic: string) => {
-    this.setState({ topic: topic.trim() });
+    this.setState({ topic });
   };
 
   isSendButtonEnabled = (message: string) => {
-    const { mandatoryTopics } = this.props;
-    const { stream, topic } = this.state;
     const { sharedData } = this.props.route.params;
 
-    if (sharedData.type !== 'text') {
-      return stream !== '' && (topic !== '' || !mandatoryTopics);
-    }
-
-    return stream !== '' && (topic !== '' || !mandatoryTopics) && message !== '';
+    return sharedData.type !== 'text' || message !== '';
```

Comment:
Hmm, this seems like a regression -- it means that when you actually have an empty topic, we'll show the send button as enabled even when that means you can't send, right?

And similarly for streams.

Reply:
> when you actually have an empty topic, we'll show the send button as enabled even when that means you can't send, right?

That's right. But I think this may actually be an improvement? It's good to tell the user specifically why they can't send a message, right, rather than having them guess?

I notice that the web app doesn't disable the send button, and waits for you to press it before giving you an alert. This is what I see just after pressing "Send":

![image](https://user-images.githubusercontent.com/22248748/140418486-ef5096ca-76a2-4b10-86d7-03d08b606045.png)

One thing that I prefer in the web app's UX, for accessibility, is that the alert persists right alongside the relevant input, even while you go and change the value of the input. (Until you dismiss the alert anyway.) When it's in a modal, like I have here in this revision, you have to keep the validation requirement in your working memory while you try to go fix the error.

But I guess I still think it's an improvement over the status quo, where there isn't a validation message at all.

Reply:
Hmm interesting. It's true, it's good to be explicit about what's missing.

Still it seems like not good UX for the "send" button to be right there brightly beckoning to you when it's not actually possible to send. I don't think that's a good UX choice in the web app either.

For a quick comparison, I just opened up Gmail on Android and went to compose a new email. It actually does both:
* When I haven't entered anything (even in the To: field), the "send" icon is grayed out.
* But if I tap it anyway, I get a little modal error dialog: "Add at least one recipient.", with an "OK" button.
* Then once I enter a valid email address, the icon gets its normal blue color.